core/src/main/scala/org/broadinstitute/dsp/workbench/welder/BackgroundTask.scala

```diff
@@ -92,7 +92,7 @@ class BackgroundTask(
         storageLinks <- storageLinksCache.get
         implicit0(tid: Ask[IO, TraceId]) <- IO(TraceId(UUID.randomUUID().toString)).map(tid => Ask.const[IO, TraceId](tid))
         _ <- storageLinks.values.toList.traverse { storageLink =>
-          if (storageLink.pattern.toString.contains(".Rmd")) {
+          if (storageLink.pattern.toString == ".*\\.Rmd" || storageLink.pattern.toString == ".*\\.R") {
```

Comment:
maybe we can have a function like
```
val filesShouldSyncRegex = ".*(.R)$|(.Rmd)$".r //put this on top of the class
def shouldSync(storageLinkPattern String): Boolean = filesShouldSyncRegex.findFirstIn(storageLinkPattern)
```

and use this function here instead.

Also add some unit tests to test out the function